### PR TITLE
Mesoshapes County Import - Round 2.0

### PR DESCRIPTION
**Pull request by stepps00**
This PR includes new admin2 (`county`) features for:

- Tunisia
- Somalia
- Somaliland
- Qatar
- Libya
- Guinea Bissau
- Eritrea
- Angola

None of the countries in this PR currently have `county` features in Who's On First - all of these features are new. No deprecating or superseding was needed for any of these countries. This will require PIP-ing features.

### CODE REVIEW

**Comment by nvkelso** [comment]
Part of https://github.com/whosonfirst-data/whosonfirst-data/issues/558.

File: data/110/875/840/3/1108758403.geojson [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+    "edtf:inception":"uuuu",
+    "geom:area":0.160014,
+    "geom:bbox":"12.2905423248,-5.06539550604,12.8251046027,-4.55689858092",
+    "geom:latitude":-4.803914,
+    "geom:longitude":12.547005,
+    "iso:country":"AO",
+    "lbl:latitude":-4.790451,
+    "lbl:longitude":12.541952,
+    "meso:adm1_alt":"",
+    "meso:adm1_loc":"Cabinda",
```
Comment (nvkelso) [inline data/110/875/840/3/1108758403.geojson]: Surprised this didn't map to [https://whosonfirst.mapzen.com/spelunker/id/85667683/#8/-5.078/12.542/](85667683) in the `"wof:belongs"` and `"wof:hierarchy"` sections.

File: data/110/875/840/5/1108758405.geojson [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+    "edtf:inception":"uuuu",
+    "geom:area":1.614371,
+    "geom:bbox":"17.2236433836,-17.837758,19.3194536972,-16.3633907969",
+    "geom:latitude":-17.055028,
+    "geom:longitude":18.1595,
+    "iso:country":"AO",
+    "lbl:latitude":-16.961746,
+    "lbl:longitude":18.143719,
+    "meso:adm1_alt":"",
+    "meso:adm1_loc":"Cuando Cubango",
```
Comment (nvkelso) [inline data/110/875/840/5/1108758405.geojson]: Ditto [85667695](https://whosonfirst.mapzen.com/spelunker/id/85667695/#6/-15.814/19.922) as region.

File: data/110/875/841/9/1108758419.geojson [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+    "meso:adm1_loc":"Luanda",
+    "meso:gadm_adm2":"Samba",
+    "meso:gadm_alt":"",
+    "meso:gaul_adm2":"Viana",
+    "meso:hasc_id":"AO.LU.SA",
+    "meso:mps_x":13.119571,
+    "meso:mps_y":-8.925525,
+    "meso:name_alt":"Samba",
+    "meso:name_en":"Samba",
+    "meso:source":"EDP",
```
Comment (nvkelso) [inline data/110/875/841/9/1108758419.geojson]: Did we add sources for all the `"meso:source"` values yet?

File: data/110/875/841/5/1108758415.geojson [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
...-8.95669460299996],[13.10544490800004,-8.95583343499987],[13.10752773300015,-8.95377731299993],[13.10833358800011,-8.95377731299993],[13.10994434399998,-8.9521389009999],[13.11330509200008,-8.95208358799994],[13.11583328199998,-8.95377731299993],[13.12583351100011,-8.95288848899992],[13.12663841200015,-8.95377731299993],[13.12749958000018,-8.95377731299993],[13.13249969500009,-8.95791625999988],[13.13580513000011,-8.95794486999995],[13.13747215300009,-8.95622253399989],[13.13994407700017,-8.95625019099998],[13.14336109200019,-8.95791625999988],[13.15205574,-8.94838905299997],[13.15369415300012,-8.94505596199997],[13.1537218090001,-8.94252777099996],[13.15458297700013,-8.94166660299993],[13.15541648900017,-8.93669414499988],[13.15622234300002,-8.93583297699996],[13.15622234300002,-8.91499996199991],[13.15708351100005,-8.91419410699996],[13.15705585500007,-8.90666675599994],[13.16083335899998,-8.90297222099997],[13.164138794,-8.90211105299994],[13.16877746600011,-8.89922237399992],[13.17291641200012,-8.8908329009999],[13.17622184800001,-8.8850002289999],[13.17622184800001,-8.88333320599992],[13.17791652700004,-8.8800277709999],[13.17912789093178,-8.87914234336904]]],"type":"Polygon"}
```
Comment (nvkelso) [inline data/110/875/841/5/1108758415.geojson]: I suspect there are some exploded multipart that should be merged here, including:

- https://github.com/whosonfirst-data/whosonfirst-data/blob/ce04aa8148131ded258f22d5a3b8a19265c2a778/data/110/875/842/3/1108758423.geojson
- https://github.com/whosonfirst-data/whosonfirst-data/blob/ce04aa8148131ded258f22d5a3b8a19265c2a778/data/110/875/842/5/1108758425.geojson
- https://github.com/whosonfirst-data/whosonfirst-data/blob/ce04aa8148131ded258f22d5a3b8a19265c2a778/data/110/875/842/1/1108758421.geojson

File: data/110/875/845/3/1108758453.geojson [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+    "wof:supersedes":[],
+    "wof:tags":[]
+},
+  "bbox": [
+    12.596221924000076,
+    -6.050472259999935,
+    12.59955596900005,
+    -6.047083377999968
+],
+  "geometry": {"coordinates":[[[12.59622192400008,-6.04913902299995],[12.59830570200006,-6.04708337799997],[12.59916687000003,-6.04708337799997],[12.59955596900005,-6.04755544699998],[12.59955596900005,-6.04836130099994],[12.59747219100007,-6.05041646999996],[12.59661102300004,-6.05047225999994],[12.59622192400008,-6.05000019099998],[12.59622192400008,-6.04913902299995]]],"type":"Polygon"}
```
Comment (nvkelso) [inline data/110/875/845/3/1108758453.geojson]: Probably another exploded multi-part feature?

**Comment by stepps00** [comment]
Update: Libya does not have administrative counties, just regions. Upcoming commit will remove county import and instead update existing region records.

File: data/110/875/841/9/1108758419.geojson (hunk 3 above)
Comment (stepps00) [inline data/110/875/841/9/1108758419.geojson]: Not yet - I am going to collect all source attributes used for recent `admin1` and `admin2` imports. Being tracked here: https://github.com/whosonfirst/whosonfirst-sources/issues/39.

File: data/110/875/840/3/1108758403.geojson (hunk 1 above)
Comment (stepps00) [inline data/110/875/840/3/1108758403.geojson]: Should these features be re-run using an updated version of the import script? Or is this something we can import as-is?

File: data/110/875/840/5/1108758405.geojson (hunk 2 above)
Comment (stepps00) [inline data/110/875/840/5/1108758405.geojson]: See comment above.

**Comment by stepps00** [comment]
Regarding https://github.com/whosonfirst-data/whosonfirst-data/pull/555#pullrequestreview-13637400: 

tk tk tk

**Comment by stepps00** [comment]
Regarding https://github.com/whosonfirst-data/whosonfirst-data/pull/555#pullrequestreview-13637881:

Qatar is OK to import as-is. As mentioned, Statoids only has HASC information for the 7 regions. https://github.com/whosonfirst-data/whosonfirst-data/issues/589 was filed to investigate admin2 HASC codes.

**Comment by stepps00** [comment]
Regarding https://github.com/whosonfirst-data/whosonfirst-data/pull/555#pullrequestreview-13640450:

https://github.com/whosonfirst-data/whosonfirst-data/issues/590 is open to track the mis-match of county features.

**Comment by stepps00** [comment]
Regarding https://github.com/whosonfirst-data/whosonfirst-data/pull/555#pullrequestreview-13639379:

Commit c531ee2 updated the counties that were in Somalia, but should actually be in Somaliland. The `parent_id` is still `-1`, but this should change when new `region` features are added for Somalia and Somaliland.

**Comment by stepps00** [comment]
The recent commits should have taken care of all outstanding issues in this branch. The branch should be ready to merge to `master`, barring any additional changes that are needed.

cc @nvkelso

File: data/110/875/840/3/1108758403.geojson (hunk 1 above)
Comment (nvkelso) [inline data/110/875/840/3/1108758403.geojson]: We can import as-is, filed https://github.com/mapzen/whosonfirst-toolbox/issues/4 for followup.

**Comment by nvkelso** [comment]
@stepps00 Please address Eritrea, then ready to merge.

**Comment by stepps00** [comment]
@nvkelso - the features called out above are actually in Angola. The original import of Angola was reverted ([here](a621ac0)), `admin2` features were dissolved in QGIS, and re-imported ([here](0d6c883)).

Sources ([1](https://en.wikipedia.org/wiki/List_of_municipalities_of_Angola), [2](http://www.statoids.com/yao.html)) show 162 and 163 `admin2` features for Angola, though the mesoshapes import contained 164 features. https://github.com/whosonfirst-data/whosonfirst-data/issues/591 was opened to address this discrepency in features counts. 

As for Eritrea, it looks like Statoids is out of date, as is the data being imported. When comparing imported features to [this source](https://en.wikipedia.org/wiki/Subdivisions_of_Eritrea), it seems there was recent restructuring that merged more than two `admin2` areas together. Opened https://github.com/whosonfirst-data/whosonfirst-data/issues/592 to address this issue. _Note: the single feature discrepency comes from Mensura, which looks to be a new county and region (HASC: ER.BA.MA)_

**Comment by nvkelso** [comment]
Thanks for the clarification, ready to merge.

**Pull request merged**